### ISSUE TITLE
[Sum/Average Loss Reduction Strategy] Why going with Sum as default ?

### ISSUE THREAD

**Issue by DEKHTIARJonathan** [tensorflow/tensorflow#31317]
TF 1.14.0 introduced the following:

> Set default loss reduction as AUTO for improving reliability of loss scaling with distribution strategy and custom training loops. AUTO indicates that the reduction option will be determined by the usage context. For almost all cases this defaults to SUM_OVER_BATCH_SIZE. When used in distribution strategy scope, outside of built-in training loops such as tf.keras compile and fit, we expect reduction value to be 'None' or 'SUM'. Using other values will raise an error. 

Source: https://github.com/tensorflow/tensorflow/releases/tag/v1.14.0

----------

Why `SUM_OVER_BATCH_SIZE` was made the default when the followings are standard practices:
- averaging the loss over one batch 
- averaging gradients over multiple ranks

Could anyone give examples of use-cases where sum would give better results than average ? And why would this made default when averaging seems far more used in the state of the art ?

**Comment by srihari-humbarwadi** [comment]
You can find it mentioned here https://www.tensorflow.org/beta/tutorials/distribute/training_loops
```
Normally, on a single machine with 1 GPU/CPU, loss is divided by the number of 
examples in the batch of input. So, how should the loss be calculated when using 
a tf.distribute.Strategy?For an example, let's say you have 4 GPU's and a batch 
size of 64.One batch of input is distributed across the replicas (4 GPUs), each 
replica getting an input of size 16.The model on each replica does a forward pass 
with its respective input and calculates the loss. Now, instead of dividing the loss
by the number of examples in its respective input (BATCH_SIZE_PER_REPLICA = 16),
the loss should be divided by the GLOBAL_BATCH_SIZE (64).
Why do this?
This needs to be done because after the gradients are calculated on each replica, 
they are synced across the replicas by summing them.
How to do this in TensorFlow? If you're writing a custom training loop, as in this 
tutorial, you should sum the per example losses and divide the sum by the 
GLOBAL_BATCH_SIZE: scale_loss = tf.reduce_sum(loss) * (1. / GLOBAL_BATCH_SIZE) 
or you can use tf.nn.compute_average_loss which takes the per example loss, optional 
sample weights, and GLOBAL_BATCH_SIZE as arguments and returns the scaled loss.
```

**Comment by jvishnuvardhan** [comment]
Automatically closing this out since I understand it to be resolved by @srihari-humbarwadi , but please let me know if I'm mistaken.Thanks!
